pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -112,18 +112,17 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
 				return err
 			}
 		}
-		// if the Pod is not dirty do nothing
+		// if the Pod is not dirty, do nothing
 		if consistent {
 			return nil
 		}
 
 		attemptedUpdate = true
 		// commit the update, retrying on conflicts
-		_, err := spc.client.Core().Pods(set.Namespace).Update(pod)
-		if err == nil {
-			return nil
+		_, updateErr := spc.client.Core().Pods(set.Namespace).Update(pod)
+		if updateErr == nil {
+			return updateErr
```

Comment:
Just realized that you also changed this one. Please revert it back to nil, when an error returned is nil, it's preferred to explicitly return nil.